src/transformers/models/llama/modeling_llama.py

```diff
@@ -825,7 +825,7 @@ def forward(
         logits = self.lm_head(hidden_states[:, slice_indices, :])
 
         loss = None
-        if labels is not None:
+        if labels is not None or kwargs.get("shift_labels", None) is not None:
```

Comment:
why not leave it to the loss function?

Reply:
We can, but it means that we have to compute the loss outside the model's `forward` with 
the user have to do

```python
from transformers import ForCausalLMLoss

shift_labels = .... 
loss = ForCausalLMLoss(logits=logits, shift_labels=shift_labels, vocab_size=model.config.vocab_size)
```

It's not too much work for users, but it is nice if we could make it easier for them (they only need to take care of preparing `shift_labels `).

Enable this API means people could perform the same workflow with `labels` and with `shift_labels` (i.e. put them in model.forward and get it from outputs).

Reply:
Also, we can't just pass `shift_labels` without passing `labels`

> loss = ForCausalLMLoss(logits=logits, shift_labels=shift_labels, vocab_size=model.config.vocab_size)

as it is a `required positional argument`. We have to do

> loss = ForCausalLMLoss(logits=logits, labels=None, shift_labels=shift_labels, vocab_size=model.config.vocab_size)

This kind of details could hide from user if we do it in modeling code.

Reply:
yeah, someone was reported using a side-effect hack - passing `labels=garbage, shift_labels=real_data` and getting the `model` to give them the loss ;)

Reply:
what I meant @ydshieh is that if we do this we need to update all models!